src/aiodynamo/credentials.py

```diff
@@ -289,6 +293,81 @@ def and_then(thing: Optional[T], mapper: Callable[[T], U]) -> Optional[U]:
     return thing
 
 
+# https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html
+@dataclass
+class AssumeRoleWithWebIdentityCredentials(MetadataCredentials):
+    ACTION = "AssumeRoleWithWebIdentity"
+    VERSION = "2011-06-15"
+
+    timeout: Timeout = 2
+    max_attempts: int = 3
+
+    base_url: URL = URL("https://sts.amazonaws.com/")
+
+    async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
+        config: dict = self._load_config()
+
+        url = self._build_url(**config)
+        response: bytes = await fetch_with_retry_and_timeout(
+            http=http,
+            max_attempts=self.max_attempts,
+            timeout=self.timeout,
+            request=Request(
+                method="GET", url=url, headers=None, body=None
+            ),
+        )
+        credentials = self._parse_aws_response(response)["Credentials"]
+        return Metadata(
+            key=Key(
+                id=credentials["AccessKeyId"],
+                secret=credentials["SecretAccessKey"],
+                token=credentials.get("Token", None),
+            ),
+            expires=parse_amazon_timestamp(credentials["Expiration"]),
+        )
+
+    def is_disabled(self) -> bool:
+        return False
+
+    def _parse_aws_response(self, response: bytes) -> dict:
+        parsed_response = xmltodict.parse(response)
+        return parsed_response["AssumeRoleWithWebIdentityResponse"]["AssumeRoleWithWebIdentityResult"]
+
+    def _build_url(self, identity_token: str, role: str, role_session_name: Optional[str]) -> str:
+        params = {
+            "WebIdentityToken": identity_token,
+            "RoleArn": role,
+            "RoleSessionName": role_session_name or f"default-role-name-{time.time()}",
```

Comment:
is that really a reasonable default? I feel like this would be better as an explicit value users have to set when using this credentials loader.

Reply:
botocore to do the same: https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L788
I decide to save this behavior for consistency between libraries. It can help to move from botocore a little bit.